### PR TITLE
Document most of `gflib` and `libpmagb`

### PR DESCRIPTION
**Pull request by citrusbolt**

### CODE REVIEW

File: payload/src/gflib/sound.c [hunk 1]
```diff
@@ -1,27 +1,28 @@
 #include "global.h"
 #include "gflib/sound.h"
 
-static u8 gUnknown_02021368;
-static bool8 isSoundVsync;
+static u8 sIsSoundVSyncPaused;
```
Comment (GriffinRichards) [inline payload/src/gflib/sound.c]: `bool8`

File: payload/include/libpmagb/agb_rom.h [hunk 2]
```diff
@@ -1,85 +1,86 @@
...
+    u8 abilityNameLength;
+    u8 typeNameLength;
+    u8 mapNameLength1;
+    u8 mapNameLength2;
+    u8 trainerClassNameLength;
+    u8 decorationNameLength;
+    u8 dexCategoryNameLength;
+    u8 endOfStringLength;
+    u8 frontierTrainerNameLength;
+    u8 easyChatWordLength;
```
Comment (GriffinRichards) [inline payload/include/libpmagb/agb_rom.h]: How were these determined?

File: payload/src/libpmagb/save.c [hunk 3]
```diff
@@ -29,71 +29,74 @@ struct SaveSector
 // Each 4 KiB flash sector contains 3968 bytes of actual data followed by a 128 byte footer
 #define SECTOR_DATA_SIZE 3968u
 #define SECTOR_FOOTER_SIZE 128
+#define SECTOR_SIGNATURE_OFFSET 0x0FF8
```
Comment (GriffinRichards) [inline payload/src/libpmagb/save.c]: `offsetof(struct SaveSector, signature)`

File: payload/include/libpmagb/agb_rom.h (hunk 2 above)
Comment (citrusbolt) [reply payload/include/libpmagb/agb_rom.h]: This was synced from my version of the struct in Heliodor.  I think the fields not originally referenced in this repo, pokeemerald, or pokefirered were referenced against the Platinum and Emerald leaks.  I can notate their provenance.

**Pull request merged**